File: Cash-Py/cash.py
# Cash - greedy algorithm

from cs50 import get_float


def main():
    change = get_positive_float("Change owed: ")
    coins = 0
    amount = round(change * 100)

    # If change >= 0.25
    while amount >= 25:
        coins += 1
        amount -= 25

    # If change >= 0.10
    while amount >= 10:
        coins += 1
        amount -= 10

    # If change >= 0.05
    while amount >= 5:
        coins += 1
        amount -= 5

    # If change >= 0.01
    while amount >= 1:
        coins += 1
        amount -= 1

    # Print the min number of coins
    print(coins)


def get_positive_float(prompt):
    while True:
        n = get_float(prompt)
        if n > 0:
            return n


main()

File: Cash/cash.c
#import <math.h>
#import <cs50.h>
#import <stdio.h>

int main(void)
{
    float change;
    do
    {
        change = get_float("Change: ");
    }
    while (change < 0);
    int amount = round(change * 100);
    int coins = 0;
    while (amount >= 25)
    {
        coins++;
        amount -= 25;
    }
    while (amount >= 10)
    {
        coins++;
        amount -= 10;
    }
    while (amount >= 5)
    {
        coins++;
        amount -= 5;
    }
    while (amount >= 1)
    {
        coins++;
        amount -= 1;
    }
    printf("%i\n", coins);
}
